### PR TITLE
GLSL shader

### PR DESCRIPTION
**Pull request by mbellew**
Make it possible to write shaders using native glsl syntax rather than in hlsl (avoid the transpile step).  Also wrote a little converter (h2g in the src/hlsl2glsl directory) for testing.  Example

```
warp_lang=glsl_330
warp_1=`shader_body {
warp_2=`    float corr = float( texsize.xy * texsize_noise_lq.zw );
warp_3=`    vec2 uv1 = vec2( vec2(uv.x - 0.5, uv.y - 0.5) );
warp_4=`    vec3 noiseVal = vec3( vec4 (0.03) * texture(sampler_noise_lq, uv * vec2 (0.3) + vec2 (vec4 (0.01) * rand_frame)) );
warp_5=`    noiseVal = vec3 (0.01);
warp_6=`    vec3 Feedback = vec3( GetBlur1(vec2 (1) - uv) );
warp_7=`    vec2 zz = vec2( uv1 * texsize.xy * vec2 (0.015) );
warp_8=`    zz = zz.yx;
warp_9=`    vec2 h1 = vec2( clamp(tan(zz.yx), vec2 ((-12)), vec2 (12)) * cos(vec2 (4 * (q2 + 1)) * zz) );
warp_10=`    uv.xy += h1 * texsize.zw * vec2 (4) * vec2 ((2 + q1));
warp_11=`    float r = float( texture(sampler_noise_hq, uv / vec2 (14)).x * 12 );
warp_12=`    vec2 uv6 = vec2( uv1 );
warp_13=`    uv6 = ((uv6)*(mat2_from_float_float_float_float(cos(r), sin(r), (-sin(r)), cos(r))));
warp_14=`    vec3 mus = vec3( 0.1 / sqrt(uv6.y) );
warp_15=`    vec3 crisp = vec3( texture(sampler_main, uv) );
warp_16=`    ret = (vec3 (0.995) * crisp + noiseVal) - vec3 (0.02);
warp_17=`}

### CODE REVIEW

File: src/libprojectM/Renderer/hlslparser/src/Engine.cpp [hunk 1]
```diff
@@ -40,7 +40,17 @@ int String_Printf(char * buffer, int size, const char * format, ...) {
 }
 
 int String_FormatFloat(char * buffer, int size, float value) {
-    return String_Printf(buffer, size, "%f", value);
+    int len = String_Printf(buffer, size, "%f", value);
+    char *p = strchr(buffer,'.');
+    if (!p)
```
Comment (revmischa) [inline src/libprojectM/Renderer/hlslparser/src/Engine.cpp]: comment?

File: presets/presets_tryptonaut/A Tribute to Martin - bombyx mori - Ft Flexi - AdamFX - StahlRegen - Krash - Rovastar -  Hd in Milk T.milk [hunk 2]
```diff
@@ -474,9 +474,9 @@ warp_27=`float3 crisp= tex2D(sampler_main,(uv1)*(.8+.2*z)+.5);
 warp_28=`
 warp_29=`float3 ret1 = crisp + mus;
 warp_30=`
-warp_31=`q25 = q24+.4; //##hier auf 1 setzen
-warp_32=`q25 = .5;
-warp_33=`ret = q25*(ret1*.96-.025) + (1-q25)* GetPixel(uv_orig);
+warp_31=`float _q25 = q24+.4; //##hier auf 1 setzen
```
Comment (revmischa) [inline presets/presets_tryptonaut/A Tribute to Martin - bombyx mori - Ft Flexi - AdamFX - StahlRegen - Krash - Rovastar -  Hd in Milk T.milk]: modifying presets to work with the parser isn't really a good option. ideally the same preset file should work the same in milkdrop and projectm.

**Comment by revmischa** [comment]
I have no idea if the q vars are read-only

**Comment by coast77777777777** [comment]
@revmischa this feature breaks many things. needs completely reworked against latest code.
@mbellew do you still have interest in developing this further?
I am closing this issue for now, please reopen for review/direction

**Comment by kblaschke** [comment]
Closing the PR due to inactivity.
The idea of adding GLSL shaders is still valid, probably using an additional key like `gl_warp*` so an author could put both variants into the preset and keep it Milkdrop-compatible.

**Pull request closed without merging**